### ISSUE TITLE
Error response content type and tests

### ISSUE THREAD

**Issue by stefanorosanelli** [bedita/bedita#1004]
Something not working as expected in error response (maybe due to cake php upgrade?)

From a JSON-API call:
- 404 errors are returned with `text/html` content type
- 500 errors (like failing db connections) are returned as HTML in body and content type

Also after #1001 
From a browser call:
- 40x and 500 errors are displayed correctly in as JSON in HTML with `debug = 0`
- same errors are displayed using cake templates with `debug=1`

All these cases must be covered by tests
Last point (browser display with `debug=1`) could be postponed
